app/src/main/java/com/arctouch/codechallenge/ui/home/HomeViewModel.kt

```diff
@@ -0,0 +1,104 @@
+package com.arctouch.codechallenge.ui.home
+
+import android.arch.lifecycle.MutableLiveData
+import com.arctouch.codechallenge.R
+import com.arctouch.codechallenge.data.repository.MoviesRepository
+import com.arctouch.codechallenge.data.model.Genre
+import com.arctouch.codechallenge.data.model.Movie
+import com.arctouch.codechallenge.data.model.MoviesResponse
+import com.arctouch.codechallenge.ui.base.BaseViewModel
+import com.arctouch.codechallenge.ui.base.SingleLiveEvent
+import com.arctouch.codechallenge.util.asyncCatching
+import com.arctouch.codechallenge.util.onFailure
+import com.arctouch.codechallenge.util.onSuccess
+
+class HomeViewModel(private val moviesRepository: MoviesRepository) : BaseViewModel(){
+
+    var currentQuery = ""
+    var currentPage = 1L
+    var totalPages = 0L
+    val genres = ArrayList<Genre>()
+    val movies = ArrayList<Movie>()
+    val enableLoadMore = MutableLiveData<Boolean>()
+    val updatedMovies = SingleLiveEvent<Boolean>()
+    var searching = false
+
+    fun refreshMovies() {
+        getMovies(1)
```

Comment:
Here you are using `getMovies()`. At first glance, it seemed to me that you were retrieving some movies, but it turned out that you were just loading them. For that reason, I suggest to rename your method to `loadMovies()` for better readability.

Reply:
Make sense.